app/models/host.rb

```diff
@@ -136,6 +136,7 @@ class Host < ApplicationRecord
   virtual_column :vmm_vendor_display,           :type => :string
   virtual_column :ipmi_enabled,                 :type => :boolean
   virtual_column :archived, :type => :boolean
+  virtual_column :has_policies?, :type => :boolean
```

Comment:
I don't think you can have a `?` in a virtual column definition (because you can't have a `?` in a real column name).  I think if you just take out the `?`, it should still work however.

Reply:
removing `?` throws an error

```
[----] F, [2020-07-10T18:13:05.182143 #13422:492805c] FATAL -- : Error caught: [ActiveModel::MissingAttributeError] missing attribute: has_policies
/home/hkataria/.rvm/gems/ruby-2.5.7/gems/activerecord-5.2.4.3/lib/active_record/attribute_methods/read.rb:77:in `block in _read_attribute'
/home/hkataria/.rvm/gems/ruby-2.5.7/gems/activemodel-5.2.4.3/lib/active_model/attribute_set.rb:48:in `block in fetch_value'
/home/hkataria/.rvm/gems/ruby-2.5.7/gems/activemodel-5.2.4.3/lib/active_model/attribute.rb:222:in `value'
/home/hkataria/.rvm/gems/ruby-2.5.7/gems/activemodel-5.2.4.3/lib/active_model/attribute_set.rb:48:in `fetch_value'
/home/hkataria/.rvm/gems/ruby-2.5.7/gems/activerecord-5.2.4.3/lib/active_record/attribute_methods/read.rb:77:in `_read_attribute'
/home/hkataria/.rvm/gems/ruby-2.5.7/gems/activerecord-5.2.4.3/lib/active_record/attribute_methods/read.rb:40:in `__temp__861637f507f6c69636965637'
/home/hkataria/dev/manageiq/app/models/miq_report/generator.rb:739:in `block in build_get_attributes_with_options'
/home/hkataria/dev/manageiq/app/models/miq_report/generator.rb:735:in `each'
/home/hkataria/dev/manageiq/app/models/miq_report/generator.rb:735:in `build_get_attributes_with_options'
/home/hkataria/dev/manageiq/app/models/miq_report/generator.rb:723:in `build_reportable_data'
/home/hkataria/dev/manageiq/app/models/miq_report/generator.rb:436:in `block in build_table'
/home/hkataria/dev/manageiq/app/models/miq_report/generator.rb:435:in `collect'
/home/hkataria/dev/manageiq/app/models/miq_report/generator.rb:435:in `build_table'
/home/hkataria/dev/manageiq/app/models/miq_report/search.rb:127:in `paged_view_search'
/home/hkataria/dev/manageiq-ui-classic/app/controllers/application_controller.rb:1224:in `get_view'
```

Reply:
yeah, then it needs the corresponding method too...you'll have to alias it.